Makefile

```diff
@@ -32,11 +36,13 @@ $(TESTS):
 	# TODO: Figure out how to name multiple CC and produce corresponding object for each
 	$(CXX) $(SRC_FLAGS) $(GTEST_FLAGS) $(PARSER_PATH)config_parser.cc server_config.cc $(TESTS).cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(LDFLAGS) -o $(TESTS)
 
-test: $(TARGET) $(TESTS)
+integration_test: $(TARGET) $(TESTS)
```

Comment:
This is named `integration_test` but actually runs the unit-tests too. Can we split these out into separate targets, and have `test` call `unit_tests` and `integration_tests`?